File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.java
class Solution {
    public int countPrefixSuffixPairs(String[] words) {
        int ans = 0;
        for(int i=0;i<words.length;i++){
            for(int j=i+1;j<words.length;j++){
                if(isPrefixAndSuffix(words[i],words[j])==true){
                    ans++;
                }
            }
        }
        return ans;
    }
    public static boolean isPrefixAndSuffix(String str1, String str2){
        int x = str1.length();
        int y = str2.length();
        if(x>y){
            return false;
        }
        int diff = str2.length()-x;
        if(str1.equals(str2.substring(0,x)) && str1.equals(str2.substring(diff,y))){
            return true;
        }else{
            return false;
        }
        
    }
}

File: 3042-count-prefix-and-suffix-pairs-i/3042-count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        def isPrefixAndSuffix(str1:str, str2:str) -> bool:
            x = len(str1)
            y = len(str2)
            if(x>y):
                return False
            if str1==str2[:x] and str1==str2[-x:]:
                return True
            else:
                return False
        ans = 0
        for i in range(len(words)):
            for j in range(i+1, len(words)):
                if isPrefixAndSuffix(words[i],words[j]):
                    ans+=1
        return ans
    
        